pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,179 @@
+        elif distribution.repository is not None:
+            repository = distribution.repository
+            repository_version = distribution.repository.latest_version()
+            # is_latest = True
+
+        # sometimes distros point at nothing?
+        else:
+            return
+
+    # extract repository from repository version if needed
+    if repository is None:
+        repository = repository_version.repository
+
+    # optimization: -must- have an AnsibleRepository for the index model
+    if not isinstance(repository, AnsibleRepository):
+        repository = AnsibleRepository.objects.filter(pk=repository.pk).first()
+        if repository is None:
+            return
+
+    # optimization: we only want to index "distributed" CVs
+    if distribution is None and not has_distribution(repository, repository_version):
+        return
+
+    if use_repository_version:
+        if CVIndex.objects.filter(repository_version=repository_version).exists():
+            return
+
+    # get all CVs in this repository version
+    cvs = repository_version.content.filter(pulp_type="ansible.collection_version").values_list(
+        "pk", flat=True
+    )
+    cvs = CollectionVersion.objects.filter(pk__in=cvs)
+
+    # get the set of signatures in this repo version
+    repo_signatures_pks = repository_version.content.filter(
+        pulp_type="ansible.collection_signature"
+    ).values_list("pk", flat=True)
+    repo_signatures = CollectionVersionSignature.objects.filter(pk__in=repo_signatures_pks)
+
+    # get the set of deprecations in this repo version
+    deprecations = repository_version.content.filter(
+        pulp_type="ansible.collection_deprecation"
+    ).values_list("pk", flat=True)
+    deprecations = AnsibleCollectionDeprecated.objects.filter(pk__in=deprecations)
+    deprecations_set = {(x.namespace, x.name) for x in deprecations}
+
+    # find all namespaces in the repo version
+    namespaces = repository_version.content.filter(pulp_type="ansible.namespace").values_list(
+        "pk", flat=True
+    )
+    namespaces = AnsibleNamespace.objects.filter(pk__in=namespaces)
+    namespaces = {x.name: x for x in namespaces}
+
+    # map out the namespace(s).name(s) for everything in the repo version
+    colset = set(cvs.values_list("namespace", "name").distinct())
+
+    # clean up any repo+namespace+name indexes that no longer live in the repo ...
+    if not use_repository_version:
+        # CVIndex.objects.filter(repository=repository, repository_version=None).exclude(
+        #    collection_version__pk__in=cvs.values_list("pk", flat=True)
+        # ).delete()
+        pass
+
+    # iterate through each collection in the repository
+    for colkey in colset:
+        namespace, name = colkey
+
+        # get all the versions for this collection
+        related_cvs = cvs.filter(namespace=namespace, name=name).only(
+            "namespace", "name", "version"
+        )
```

Comment:
```suggestion
        related_cvs = cvs.filter(namespace=namespace, name=name).only("version")
```
These optimizations are getting pedantic, but `version` is the only field we actually need.